lib/money-rails/active_record/monetizable.rb

```diff
@@ -104,6 +104,11 @@ def monetize(*fields)
               end
             end
 
+            # Check if the options[:as] parameter is not the same as subunit_name
+            # which would result in stack overflow
+            if name == subunit_name
```

Comment:
You definitely wanna raise this as soon as you can, maybe before the `track_monetized_attribute` call